src/proxy-ui-api/frontend/src/views/Clients/ServiceClients/ServiceClientAccessRights.vue

```diff
@@ -99,36 +108,83 @@ export default Vue.extend({
     return {
       accessRights: [] as AccessRight[],
       serviceClient: {} as ServiceClient,
+      isAddServiceDialogVisible: false as boolean,
+      serviceDescriptions: [] as ServiceDescription[],
+      services: [] as Service[],
     };
   },
   methods: {
     fetchData() {
 
+      this.fetchAccessRights();
+
       api
         .get(`/clients/${this.id}/service-clients/${this.serviceClientId}`)
         .then( (response: any) => this.serviceClient = response.data)
-        .catch( (error: any) =>
-          this.$store.dispatch('showError', error));
+        .catch( (error: any) => this.$store.dispatch('showError', error));
 
+      api
+        .get(`/clients/${this.id}/service-descriptions`)
+        .then( (response: any) => {
+          this.serviceDescriptions = response.data;
+        })
+        .catch( (error: any) => this.$store.dispatch('showError', error));
+    },
+    fetchAccessRights() {
       api
         .get(`/clients/${this.id}/service-clients/${this.serviceClientId}/access-rights`)
         .then( (response: any) => this.accessRights = response.data)
-        .catch( (error: any) =>
-          this.$store.dispatch('showError', error));
-
+        .catch( (error: any) => this.$store.dispatch('showError', error));
     },
-    close() {
+    close(): void {
       this.$router.go(-1);
     },
-    remove() {
-      // NOOP
+    addService(accessRights: AccessRight[]): void {
+      this.hideAddService();
+
+      const accessRightsObject: AccessRights = { items: accessRights };
+      api
+        .post(`/clients/${this.id}/service-clients/${this.serviceClientId}/access-rights`, accessRightsObject)
+        .then( (response: any) => {
+          this.$store.dispatch('showSuccess', 'serviceClients.addServiceClientAccessRightSuccess');
+          this.fetchAccessRights();
+        })
+        .catch( (error: any) => this.$store.dispatch('showError', error));
     },
-    addService() {
+    hideAddService(): void {
+      this.isAddServiceDialogVisible = false;
+    },
+    remove(): void {
       // NOOP
     },
-    removeAll() {
+    showAddServiceDialog(): void {
+      this.isAddServiceDialogVisible = true;
+    },
+    removeAll(): void {
       // NOOP
     },
+    serviceCandidates(): any[] {
```

Comment:
I suppose that there is no interface in the types for the object that this method returns.
Would it make sense to create an interface that extends AccessRight just for the ui ?
If so, then this might be a good place for it
https://github.com/nordic-institute/X-Road/blob/develop/src/proxy-ui-api/frontend/src/ui-types.ts

Reply:
No such file in my branch so I can't use it. Added a ServiceCandidate-type directly to ServiceClientAcccessRights.vue